dftimewolf/lib/exporters/timesketch_tam.py

```diff
@@ -0,0 +1,250 @@
+    self.sketch = None
+
+    # Check that we have a timesketch session.
+    if not (self.timesketch_api or self.timesketch_api.session):
+      message = 'Could not connect to Timesketch server'
+      self.ModuleError(message, critical=True)
+
+    # If no sketch ID is provided through the CLI, attempt to get it from
+    # attributes
+    if not self.sketch_id:
+      self.sketch_id = self._GetSketchIDFromAttributes()
+
+    # If we have a sketch ID, check that we can write to it and cache it.
+    if self.sketch_id:
+      self.sketch = self.timesketch_api.get_sketch(self.sketch_id)
+      if 'write' not in self.sketch.my_acl:
+        self.ModuleError(
+            'No write access to sketch ID {0:d}, aborting'.format(
+                self.sketch_id),
+            critical=True)
+      self.state.AddToCache('timesketch_sketch', self.sketch)
+      self.sketch_id = self.sketch.id
+
+    if analyzers and isinstance(analyzers, (tuple, list)):
+      self._analyzers = analyzers
+
+  def _CreateSketch(self, incident_id: None=None) -> ts_sketch.Sketch:
+    """Creates a new Timesketch sketch.
+
+    Args:
+      incident_id (str): Incident ID to use sketch description.
+
+    Returns:
+      timesketch_api_client.Sketch: An instance of the sketch object.
+    """
+    if incident_id:
+      sketch_name = 'Sketch for incident ID: ' + incident_id
+    else:
+      sketch_name = 'Untitled sketch'
+    sketch_description = 'Sketch generated by dfTimewolf'
+
+    sketch = self.timesketch_api.create_sketch(
+        sketch_name, sketch_description)
+    self.sketch_id = sketch.id
+    if incident_id:
+      sketch.add_attribute(
+          'incident_id', incident_id, ontology='text')
+    self.state.AddToCache('timesketch_sketch', sketch)
+
+    return sketch
+
+  def _WaitForTimelines(self) -> None:
+    """Waits for all timelines in a sketch to be processed."""
+    time.sleep(5)  # Give Timesketch time to populate recently added timelines.
+    sketch = self.timesketch_api.get_sketch(self.sketch_id)
+    timelines = sketch.list_timelines()
+    while True:
+      if all(tl.status in ['fail', 'ready', 'timeout', 'archived']
+             for tl in timelines):
+        break
+      time.sleep(10)
+
+  def _GetSketchIDFromAttributes(self) -> int:
+    """Attempts to retrieve a Timesketch ID from ticket attributes.
+
+    Returns:
+      int: the sketch idenifier, or 0 if one was not available.
+    """
+    attributes = self.state.GetContainers(containers.TicketAttribute)
+    for attribute in attributes:
+      if attribute.name == self._SKETCH_ATTRIBUTE_NAME:
+        sketch_match = re.search(r'sketch/(\d+)/', attribute.value)
+        if sketch_match:
+          sketch_id = int(sketch_match.group(1), 10)
+          return sketch_id
+    return 0
+
+  def Process(self, container: containers.File) -> None:
+    """Executes a Timesketch export."""
+
+    recipe_name = self.state.recipe.get('name', 'no_recipe')
+
+    description = container.name
```

Comment:
Looks like you're using both description and container.name below, maybe use just one?